src/plugins/credential_management/public/components/create_credential_wizard/components/header/add_form.tsx

```diff
@@ -0,0 +1,187 @@
+      this.setState({ nameShowErrors: true });
+    }
+    if (
+      !password &&
+      credentialMaterialsType === Credential.CredentialMaterialsType.UsernamePasswordType
+    ) {
+      validationByField.push('Password field should not be empty');
+      this.setState({ nameShowErrors: true });
+    }
+
+    if (credentialMaterialsType === 'no_auth') {
+      this.setState({ nameShowErrors: false });
+    }
+
+    if (validationByField.length) {
+      this.setState({ showErrors: true, errors: validationByField });
+      return;
+    }
+
+    this.setState({ showErrors: false, errors: [] });
+    this.props.onLoading();
+    const { savedObjects }: any = this.props;
+
+    try {
+      await savedObjects.client.create('credential', {
+        title: this.state.credentialName,
+        description: this.state.description,
+        credentialMaterials: {
+          credentialMaterialsType: this.state.credentialMaterialsType,
+          credentialMaterialsContent: {
+            username: this.state.userName,
+            password: this.state.password,
+          },
+        },
+      });
+      this.props.onHispush();
+    } catch (e) {
+      const createCredentialFailMsg = (
+        <FormattedMessage
+          id="credentialManagement.createCredential.loadCreateCredentialFailMsg"
+          defaultMessage="The credential saved object creation failed with some errors. Please configure data_source.enabled and try it again."
+        />
+      );
+      this.setState((prevState: any) => ({
+        toasts: prevState.toasts.concat([
+          {
+            title: createCredentialFailMsg,
+            id: createCredentialFailMsg.props.id,
+            color: 'warning',
+            iconType: 'alert',
+          },
+        ]),
+      }));
+    }
+    this.props.onCancelLoading();
+  };
+
+  render() {
+    const { showErrors, errors, nameShowErrors } = this.state;
+
+    const options = [
+      {
+        value: Credential.CredentialMaterialsType.UsernamePasswordType,
+        text: 'Username and Password Credential',
+      },
+      {
+        value: Credential.CredentialMaterialsType.NoAuth,
+        text: 'No Auth',
+      },
+    ];
+
+    return (
+      <EuiForm component="form" error={errors} isInvalid={showErrors}>
+        <EuiFormRow label="Credential Title" isInvalid={showErrors}>
+          <EuiFieldText
+            placeholder="Your Credential Title"
+            value={this.state.credentialName || ''}
+            onChange={(e) => this.setState({ credentialName: e.target.value })}
+            isInvalid={showErrors}
+          />
+        </EuiFormRow>
+
+        <EuiFormRow label="Credential Description" isInvalid={showErrors}>
+          <EuiFieldText
+            placeholder="Your Credential Description"
+            value={this.state.description || ''}
+            onChange={(e) => this.setState({ description: e.target.value })}
+            isInvalid={showErrors}
```

Comment:
Let's say,
1. I do not enter few required fields (title) and click on create > it will show validation errors
2. When I start typing in title field, it should do validation to avoid confusion to the user.